src/SixLabors.Fonts/GlyphSubstitutionCollection.cs

```diff
@@ -276,15 +291,30 @@ public void Replace(int index, int count, ushort glyphId)
     {
         // Remove the glyphs at each index.
         int codePointCount = 0;
+        CodePoint codePoint = default;
         for (int i = count; i > 0; i--)
         {
             int match = index + i;
             codePointCount += this.glyphs[match].Data.CodePointCount;
+            CodePoint currentCodePoint = this.glyphs[match].Data.CodePoint;
+            if (!UnicodeUtility.IsDefaultIgnorableCodePoint((uint)codePoint.Value) || UnicodeUtility.ShouldRenderWhiteSpaceOnly(codePoint))
```

Comment:
When substituting a ligation, we need to ensure that the remaining codepoint is renderable.